flink-runtime/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperatorV2.java

```diff
@@ -220,7 +220,7 @@ public void initializeState(StreamTaskStateInitializer streamTaskStateManager)
                         isUsingCustomRawKeyedState());
 
         stateHandler = new StreamOperatorStateHandler(context, getExecutionConfig(), cancelables);
-        timeServiceManager = context.internalTimerServiceManager();
+        timeServiceManager = context.asyncInternalTimerServiceManager();
```

Comment:
I think this is not true. We'd better keep `context.internalTimerServiceManager` here, and re-assign `timeServiceManager` in `AbstractAsyncStateStreamOperatorV2` with `asyncInternalTimerServiceManager`.